crates/data-generator/src/blocks/output.rs

```diff
@@ -0,0 +1,628 @@
+                            #match_arms
+                            _ => Err(super::UnknownId(id))?,
+                        })
+                    }
+                }
+            });
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct BlockStateInst {
+        pub properties: Vec<BlockStateProperty>,
+    }
+
+    impl From<&State> for BlockStateInst {
+        fn from(block_state: &State) -> Self {
+            let properties = block_state
+                .properties
+                .iter()
+                .map(|block_prop| BlockStateProperty {
+                    field_name: block_prop.field.to_ident(),
+                    enum_name: block_prop.prop_enum.to_ident(),
+                    var_name: block_prop.variant.to_ident(),
+                })
+                .collect();
+
+            BlockStateInst { properties }
+        }
+    }
+
+    impl ToTokens for BlockStateInst {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            // "Ladder"
+            tokens.append("Block".to_ident());
+
+            if !self.properties.is_empty() {
+                let mut props = TokenStream::new();
+                // "facing: Facing::North, waterlogged: Waterlogged::True,"
+                self.properties
+                    .iter()
+                    .for_each(|prop| prop.to_tokens(&mut props));
+
+                // "LargeAmethystBud { facing: Facing::North, waterlogged: Waterlogged::True, }"
+                tokens.append(Group::new(Delimiter::Brace, props))
+            }
+        }
+    }
+
+    #[derive(Debug)]
+    pub struct BlockStateProperty {
+        pub field_name: Ident, // instrument
+        pub enum_name: Ident,  // NoteBlock
+        pub var_name: Ident,   // IronXylophone
+    }
+
+    impl ToTokens for BlockStateProperty {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let field_name = &self.field_name;
+            let enum_name = &self.enum_name;
+            let var_name = &self.var_name;
+
+            // "instrument: NoteBlock::IronXylophone,"
+            tokens.extend(quote!(#field_name: #enum_name::#var_name,))
+        }
+    }
+
+    impl ToTokens for Id {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            tokens.append(Literal::u32_unsuffixed(self.0))
+        }
+    }
+}
+
+mod prop_enum {
+    use proc_macro2::{Delimiter, Group, Literal, Punct, Spacing, TokenStream};
+    use quote::{quote, ToTokens, TokenStreamExt as _};
+
+    use super::StringExt as _;
+    use crate::blocks::model::BlockProperty;
+
+    pub struct PropertyEnum {
+        pub prop_name: String, // Powered
+        pub prop_vars: Vec<PropertyEnumVariant>,
+        pub ordered: bool,
+    }
+
+    impl From<&BlockProperty> for PropertyEnum {
+        fn from(block_prop: &BlockProperty) -> Self {
+            let prop_name_pc = block_prop.name_pc.clone();
+            let ordered = block_prop.is_num();
+
+            let prop_variants: Vec<PropertyEnumVariant> = block_prop
```

Comment:
```suggestion
            let prop_variants: Vec<_> = block_prop
```

even better:

```suggestion
            let prop_variants = block_prop
```

since rustc knows the type of `prop_variants` based on its following usage.

Reply:
Obsolete as of 4115128d4ceb2f4c64f815c965bd5728e6fff786.